File: packages/create-react-native-library/templates/java-view-new/android/src/main/java/com/{%- project.package_dir %}/{%- project.name %}ViewManager.java
package com.<%- project.package %>;

import android.graphics.Color;

import androidx.annotation.Nullable;

import com.facebook.react.module.annotations.ReactModule;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewManagerDelegate;
import com.facebook.react.uimanager.annotations.ReactProp;
import com.facebook.react.viewmanagers.<%- project.name -%>ViewManagerDelegate;
import com.facebook.react.viewmanagers.<%- project.name -%>ViewManagerInterface;

@ReactModule(name = <%- project.name -%>ViewManager.NAME)
public class <%- project.name -%>ViewManager extends SimpleViewManager<<%- project.name -%>View> implements <%- project.name -%>ViewManagerInterface<<%- project.name -%>View> {

  public static final String NAME = "<%- project.name -%>View";

  private final ViewManagerDelegate<<%- project.name -%>View> mDelegate;

  public <%- project.name -%>ViewManager() {
    mDelegate = new <%- project.name -%>ViewManagerDelegate(this);
  }

  @Nullable
  @Override
  protected ViewManagerDelegate<<%- project.name -%>View> getDelegate() {
    return mDelegate;
  }

  @Override
  public String getName() {
    return NAME;
  }

  @Override
  public <%- project.name -%>View createViewInstance(ThemedReactContext context) {
    return new <%- project.name -%>View(context);
  }

  @Override
  @ReactProp(name = "color")
  public void setColor(<%- project.name -%>View view, String color) {
    view.setBackgroundColor(Color.parseColor(color));
  }
}

File: packages/create-react-native-library/templates/kotlin-view-new/android/src/main/java/com/{%- project.package_dir %}/{%- project.name %}ViewManager.kt
package com.<%- project.package %>

import android.graphics.Color
import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewManagerDelegate
import com.facebook.react.uimanager.annotations.ReactProp
import com.facebook.react.viewmanagers.<%- project.name -%>ViewManagerInterface
import com.facebook.react.viewmanagers.<%- project.name -%>ViewManagerDelegate

@ReactModule(name = <%- project.name -%>ViewManager.NAME)
class <%- project.name -%>ViewManager : SimpleViewManager<<%- project.name -%>View>(),
  <%- project.name -%>ViewManagerInterface<<%- project.name -%>View> {
  private val mDelegate: ViewManagerDelegate<<%- project.name -%>View>

  init {
    mDelegate = <%- project.name -%>ViewManagerDelegate(this)
  }

  override fun getDelegate(): ViewManagerDelegate<<%- project.name -%>View>? {
    return mDelegate
  }

  override fun getName(): String {
    return NAME
  }

  public override fun createViewInstance(context: ThemedReactContext): <%- project.name -%>View {
    return <%- project.name -%>View(context)
  }

  @ReactProp(name = "color")
  override fun setColor(view: <%- project.name -%>View?, color: String?) {
    view?.setBackgroundColor(Color.parseColor(color))
  }

  companion object {
    const val NAME = "<%- project.name -%>View"
  }
}
